smack-experimental/src/main/java/org/jivesoftware/smackx/httpfileupload/HttpFileUploadManager.java

```diff
@@ -494,11 +497,12 @@ private void upload(InputStream iStream, long fileSize, Slot slot, UploadProgres
             case HttpURLConnection.HTTP_NO_CONTENT:
                 break;
             default:
-                throw new IOException("Error response " + status + " from server during file upload: "
-                                + urlConnection.getResponseMessage() + ", file size: " + fileSize + ", put URL: "
-                                + putUrl);
+                throw new UnsuccessfulUploadException(status, urlConnection.getResponseMessage(), fileSize, putUrl);
```

Comment:
I am not super happy with the name of the exception, but you know: naming things is hard. But what about `HttpUploadErrorException`, since it contains an HTTP error code?

Reply:
Sure, though I liked my functional descriptive ones, this is more factual indeed